protocol/parser/parser.go

```diff
@@ -107,25 +107,34 @@ func ParseBlockFromParams(rpcInput RPCInput, blockParser spectypes.BlockParser)
 	return rpcInput.ParseBlock(resString)
 }
 
-// this function returns the block that was requested,
-func ParseBlockFromReply(rpcInput RPCInput, blockParser spectypes.BlockParser) (int64, error) {
+func ParseFromReply(rpcInput RPCInput, blockParser spectypes.BlockParser) (string, error) {
 	result, err := Parse(rpcInput, blockParser, PARSE_RESULT)
 	if err != nil || result == nil {
-		return spectypes.NOT_APPLICABLE, err
+		return "", err
 	}
 
-	blockstr, ok := result[0].(string)
+	response, ok := result[0].(string)
 	if !ok {
-		return spectypes.NOT_APPLICABLE, errors.New("block number is not string parseable")
+		return "", errors.New("chain ID is not string parseable")
```

Comment:
this error is wrong since this parser is generic for blocks too

Reply:
fixed 365de6df9cbd0eda56a97a1ecde6878189fea590